service/history/execution/mutable_state_builder_methods_signal_test.go

```diff
@@ -0,0 +1,181 @@
+package execution
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/uber/cadence/common/persistence"
+	"github.com/uber/cadence/common/types"
+	"github.com/uber/cadence/service/history/constants"
+	"testing"
+)
+
+func Test__IsSignalRequested(t *testing.T) {
+	mb := testMutableStateBuilder(t)
```

Comment:
I think this will make the two cases dependend, so if the signalNotFound alters `mb` then signal found will fail due to this? 

Maybe it's better to have this line in both tests?

Reply:
atleast in this method, there is no writes and its only a lookup so I decided to reuse it.